hpx/util/invoke_fused.hpp

```diff
@@ -68,26 +68,36 @@ namespace hpx { namespace util
         }
     }
 
-    template <typename F, typename Tuple>
+    /// \copydoc invoke_fused
+    ///
+    /// \tparam R The result type of the function when it's called
+    ///           with the content of the given sequenced type.
+    template <typename R, typename F, typename Tuple>
     HPX_HOST_DEVICE HPX_FORCEINLINE
-    typename detail::fused_result_of<F&&(Tuple&&)>::type
-    invoke_fused(F&& f, Tuple&& t)
+    R invoke_fused_r(F&& f, Tuple&& t)
     {
-        typedef typename detail::fused_result_of<F&&(Tuple&&)>::type R;
-
         return detail::invoke_fused_impl<R>(
             std::forward<F>(f), std::forward<Tuple>(t),
             typename detail::fused_index_pack<Tuple>::type());
     }
 
-    template <typename R, typename F, typename Tuple>
+    /// Invokes the given functional type f with the content of
+    /// the sequenced type t (tuples, pairs)
+    ///
+    /// \returns The result of the function when it's called with
+    ///          the content of the given sequenced type.
+    ///
+    /// \note This is function is similar to `std::apply` (C++17)
+    template <typename F, typename Tuple>
     HPX_HOST_DEVICE HPX_FORCEINLINE
-    R invoke_fused_r(F&& f, Tuple&& t)
+    typename detail::fused_result_of<F&&(Tuple&&)>::type
+    invoke_fused(F&& f, Tuple&& t)
     {
-        return detail::invoke_fused_impl<R>(
-            std::forward<F>(f), std::forward<Tuple>(t),
-            typename detail::fused_index_pack<Tuple>::type());
+        typedef typename detail::fused_result_of<F&&(Tuple&&)>::type R;
+
+        return invoke_fused_r<R>(std::forward<F>(f), std::forward<Tuple>(t));
     }
```

Comment:
Well, thanks for pointing that out.

I didn't intend to apply similar changes, just spotted that `invoke_fused` isn't self explaining
thus I wanted to add documentation to it.
`invoke` (and its underlying implementation) seems good for me since it's self-explanatory.